app/controllers/order_detail_stored_files_controller.rb

```diff
@@ -48,19 +48,23 @@ def upload_order_file
         redirect_to(order_path(@order))
       end
     else
-      flash.now[:error] = "An error was encountered while uploading the Order File" # TODO: I18n
+      flash.now[:error] = text('upload_order_file.error')
       render :order_file
     end
   end
 
   # GET /orders/:order_id/order_details/:order_detail_id/remove_order_file
   def remove_order_file
-    if @order_detail.stored_files.template_result.all?(&:destroy)
-      flash[:notice] = "The uploaded Order File has been deleted successfully" # TODO: I18n
+    if @order_detail.new?
+      if @order_detail.stored_files.template_result.all?(&:destroy)
```

Comment:
I think it's fine to leave as is. Although, technically `destroy` also returns the single object that was just destroyed.

I think it's probably fine to leave as is. I can't think of where this wouldn't work. And really there's only ever one of these files at a time.